test/distributed/dd_matrix_kernels.cpp

```diff
@@ -0,0 +1,269 @@
+#include <gtest/gtest.h>
+
+#include <ginkgo/core/base/device_matrix_data.hpp>
+#include <ginkgo/core/base/executor.hpp>
+
+#include "core/test/utils.hpp"
+#include "core/test/utils/assertions.hpp"
+#include "test/utils/common_fixture.hpp"
+
+
+using comm_index_type = gko::experimental::distributed::comm_index_type;
+
+
+template <typename ValueLocalGlobalIndexType>
+class DdMatrix : public CommonTestFixture {
+protected:
+    using value_type = typename std::tuple_element<
+        0, decltype(ValueLocalGlobalIndexType())>::type;
+    using local_index_type = typename std::tuple_element<
+        1, decltype(ValueLocalGlobalIndexType())>::type;
+    using global_index_type = typename std::tuple_element<
+        2, decltype(ValueLocalGlobalIndexType())>::type;
+
+    DdMatrix() : engine(42) {}
+
+    void validate(
+        gko::ptr_param<const gko::experimental::distributed::Partition<
+            local_index_type, global_index_type>>
+            row_partition,
+        gko::ptr_param<const gko::experimental::distributed::Partition<
+            local_index_type, global_index_type>>
+            col_partition,
+        gko::ptr_param<const gko::experimental::distributed::Partition<
+            local_index_type, global_index_type>>
+            d_row_partition,
+        gko::ptr_param<const gko::experimental::distributed::Partition<
+            local_index_type, global_index_type>>
+            d_col_partition,
+        gko::device_matrix_data<value_type, global_index_type> input)
+    {
+        gko::device_matrix_data<value_type, global_index_type> d_input{exec,
+                                                                       input};
+        for (comm_index_type part = 0; part < row_partition->get_num_parts();
+             ++part) {
+            gko::array<global_index_type> non_local_row_idxs{ref};
+            gko::array<global_index_type> non_local_col_idxs{ref};
+            gko::array<global_index_type> d_non_local_row_idxs{exec};
+            gko::array<global_index_type> d_non_local_col_idxs{exec};
+
+            gko::kernels::reference::distributed_dd_matrix::
+                filter_non_owning_idxs(ref, input, row_partition.get(),
+                                       col_partition.get(), part,
+                                       non_local_row_idxs, non_local_col_idxs);
+            gko::kernels::GKO_DEVICE_NAMESPACE::distributed_dd_matrix::
+                filter_non_owning_idxs(
+                    exec, d_input, d_row_partition.get(), d_col_partition.get(),
+                    part, d_non_local_row_idxs, d_non_local_col_idxs);
+
+            GKO_ASSERT_ARRAY_EQ(d_non_local_row_idxs, non_local_row_idxs);
+            GKO_ASSERT_ARRAY_EQ(d_non_local_col_idxs, non_local_col_idxs);
+        }
+    }
+
+    std::default_random_engine engine;
+};
+
+TYPED_TEST_SUITE(DdMatrix, gko::test::ValueLocalGlobalIndexTypes,
+                 TupleTypenameNameGenerator);
+
+
+TYPED_TEST(DdMatrix, BuildsEmptyIsSameAsRef)
```

Comment:
This test isn't building anything, it tests the filter, so maybe rename it (and the other tests):
```suggestion
TYPED_TEST(DdMatrix, FiltersEmptyIsSameAsRef)
```